pkg/storage/kubernetes.go

```diff
@@ -154,6 +154,80 @@ func (i *KubernetesIPAM) Close() error {
 	return nil
 }
 
+// KubernetesOverlappingRangeStore represents a OverlappingRangeStore interface
+type KubernetesOverlappingRangeStore struct {
+	client      client.Client
+	containerID string
+	namespace   string
+}
+
+// GetOverlappingRangeStore returns a clusterstore interface
+func (i *KubernetesIPAM) GetOverlappingRangeStore() (OverlappingRangeStore, error) {
+	return &KubernetesOverlappingRangeStore{i.client, i.containerID, i.namespace}, nil
+}
+
+// IsAllocatedInOverlappingRange checks for IP addresses to see if they're allocated cluster wide, for overlapping ranges
+func (c *KubernetesOverlappingRangeStore) IsAllocatedInOverlappingRange(ctx context.Context, ip net.IP) (bool, error) {
+
+	// IPv6 doesn't make for valid CR names, so normalize it.
+	normalizedip := strings.ReplaceAll(fmt.Sprint(ip), "::", "-")
```

Comment:
(Might be stupid q, but let me ask...) in ipv6 case, how ':' is replaced, such ash "2001::2:3:4/64"?

Reply:
Nope, not a stupid question at all! I think this is a problem.

Reply:
Added a test so I could improve this functionality.... fixed this one, also found a bug elsewhere in the code for IPv6! Thank you Tomo!!!